web_poet/page_inputs/http.py

```diff
@@ -305,3 +302,14 @@ def _auto_detect_fun(self, body: bytes) -> Optional[str]:
             except UnicodeError:
                 continue
             return resolve_encoding(enc)
+
+
+def request_fingerprint(req: HttpRequest) -> str:
+    """Return the fingerprint of the request."""
+    fp = sha1()
+    fp.update(req.method.encode() + b"\n")
+    fp.update(canonicalize_url(str(req.url)).encode())
+    fp.update(str(req.url).encode() + b"\n")
+    for name, value in sorted(req.headers.items()):
```

Comment:
What's the behavior of .items() when a header has multiple values for the same name?

Reply:
["View contains all items, multiple items can have the same key."](https://multidict.aio-libs.org/en/stable/multidict.html#multidict.MultiDict.items)